InteractionBase/src/handlers/GenericHandler.cpp

```diff
@@ -153,13 +154,55 @@ void GenericHandler::showCommandPrompt(Visualization::Item* commandReceiver, QSt
 {
 	if (commandPrompt_ && commandPrompt_->commandReceiver() == commandReceiver)
 	{
-		commandPrompt_->showPrompt(initialCommandText);
+        commandPrompt_->showPrompt(initialCommandText);
+        this->showCommandMenu(commandReceiver);
 	}
 	else
 	{
-		removeCommandPrompt();
-		commandPrompt_ = new CommandPrompt(commandReceiver, initialCommandText);
-	}
+        removeCommandPrompt();
+        commandPrompt_ = new CommandPrompt(commandReceiver, initialCommandText);
+        this->showCommandMenu(commandReceiver);
+	}
+}
+
+void GenericHandler::showCommandMenu(Visualization::Item* commandReceiver)
+{
+    QList<AutoCompleteEntry*> entries;
+    for (auto* command : commands())
+    {
+        if (command->canBeUsedInMenu())
+        {
+            //We find the first parent where it can be interpreted
```

Comment:
So, I feel like we need to have a single mechanism that determines what commands are available in a given context. This should be used by both the regular command prompt and by the menu. I can help with something like that as I feel it's not really part of your project.